### PR TITLE
refactor: manifest config

### PR DESCRIPTION
**Pull request by Andrei997**
**Goals:**

- merges `manifest.yml` into `config.yml` when generating executor with `jina hub new`
- changes documentation for manifest configuration, which is now present into `config.yml`
- [x] check and update documentation. See [guide](https://github.com/jina-ai/jina/CONTRIBUTING.md#documentation-guidelines) and ask the team.

### CODE REVIEW

File: docs/fundamentals/executor/yaml-spec.md [hunk 1]
```diff
@@ -20,6 +20,12 @@ py_modules:
 metas:
   name: Indexer
   description: Indexes all documents
+manifest:
+  manifest_version: 1
+  name: MyIndexer
+  description: Indexes all documents
+  url: https://github.com/janedoe/indexer
+  keywords: ["indexer", "executor"]
```
Comment (hanxiao) [inline docs/fundamentals/executor/yaml-spec.md:28]: this looks wrong as it is not updated

Comment (Andrei997) [reply docs/fundamentals/executor/yaml-spec.md]: Fixed in the latest commit

**Comment by Andrei997** [comment]
NOTE: this one should be merged together with, or after https://github.com/jina-ai/hubble/pull/558 is merged since they address the same issue

**Comment by samsja** [comment]
One question regarding the pr : is this backward compatible with old Executor that still use the manifest.yml ?

**Comment by Andrei997** [comment]
Yes the PR is backwards compatible, only changing documentation and executors created using `jina hub new`.

**Review by mapleeit — APPROVED** [review]
LGTM

**Review by mapleeit — APPROVED** [review]

**Pull request merged**